### ISSUE TITLE
Use of -- in dirname invocations

### ISSUE THREAD

**Issue by djmattyg007** [koalaman/shellcheck#2157]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash -e

cd "$(dirname "$0")/.."
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

Shellcheck should warn that there is no `--` between `dirname` and `"$0"`. Failure to do this could unintentionally cause `"$0"` to be treated as a flag to `dirname`, rather than an actual directory name.

**Comment by guillaume-d** [comment]
Looks like a duplicate of good old #314 to me, which has much more details about the finer points.
